cmd/api-resource-collector/scap.go

```diff
@@ -0,0 +1,290 @@
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/compliance-operator/pkg/utils"
+)
+
+const (
+	contentFileTimeout = 3600
+)
+
+// For OpenSCAP content as an XML data stream. Implements ResourceFetcher.
+type scapContentDataStream struct {
+	// Client for Gets
+	client *kubernetes.Clientset
+	// Staging objects
+	dataStream *utils.XMLDocument
+	resources  map[string]string
+	found      map[string][]byte
+}
+
+func (c *scapContentDataStream) LoadSource(path string) error {
+	f, err := openNonEmptyFile(path)
+	if err != nil {
+		return err
+	}
+	// #nosec
+	defer f.Close()
+	xml, err := parseContent(f)
+	if err != nil {
+		return err
+	}
+	c.dataStream = xml
+	return nil
+}
+
+func parseContent(f *os.File) (*utils.XMLDocument, error) {
+	return utils.ParseContent(bufio.NewReader(f))
+}
+
+// Returns the file, but only after it has been created by the other init container.
+// This avoids a race.
+func openNonEmptyFile(filename string) (*os.File, error) {
+	readFileTimeoutChan := make(chan *os.File, 1)
+
+	// gosec complains that the file is passed through an evironment variable. But
+	// this is not a security issue because none of the files are user-provided
+	cleanFileName := filepath.Clean(filename)
+
+	go func() {
+		for {
+			// Note that we're cleaning the filename path above.
+			// #nosec
+			file, err := os.Open(cleanFileName)
+			if err == nil {
+				fileinfo, err := file.Stat()
+				// Only try to use the file if it already has contents.
+				if err == nil && fileinfo.Size() > 0 {
+					readFileTimeoutChan <- file
+				}
+			} else if !os.IsNotExist(err) {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+			time.Sleep(1 * time.Second)
+		}
+	}()
+
+	select {
+	case file := <-readFileTimeoutChan:
+		fmt.Printf("File '%s' found, using.\n", filename)
+		return file, nil
+	case <-time.After(time.Duration(contentFileTimeout) * time.Second):
+		fmt.Println("Timeout. Aborting.")
+		os.Exit(1)
+	}
+
+	// We shouldn't get here.
+	return nil, nil
+}
+
+func (c *scapContentDataStream) FigureResources(profile string) error {
+	paths := getResourcePaths(c.dataStream, profile)
+	if len(paths) == 0 {
+		return fmt.Errorf("no checks found in datastream")
+	}
+	c.resources = paths
+	return nil
+}
+
+const (
+	endPointTag    = "ocp-api-endpoint"
+	endPointTagEnd = endPointTag + "\">"
+	dumpTag        = "ocp-dump-location"
+	dumpTagEnd     = dumpTag + "\"><sub idref=\"xccdf_org.ssgproject.content_value_ocp_data_root\" use=\"legacy\" />"
+	codeTag        = "</code>"
+)
+
+// getPathsFromRuleWarning finds the API endpoint and save path from in. The expected structure is:
+//
+//  <warning category="general" lang="en-US"><code class="ocp-api-endpoint">/apis/config.openshift.io/v1/oauths/cluster
+//  </code><code class="ocp-dump-location"><sub idref="xccdf_org.ssgproject.content_value_ocp_data_root" use="legacy" />
+//  /idp.yml</code>file.</warning>
+//
+// We return both values or none at all.
+func getPathsFromWarningXML(in string) (string, string) {
```

Comment:
Now that I've had a better chance at reading the XCCDF standard, I commented on the patch that adds the YAML checks to the content repo. Not very keen on using a warning here, let's see what the OpenSCAP team says.